### PR TITLE
Implement `prepare_refund()`

### PR DESCRIPTION
**Pull request by danielgranhao**
Addresses #526.

### CODE REVIEW

File: libs/sdk-core/src/swap.rs [hunk 1]
```diff
@@ -434,6 +434,31 @@ impl BTCReceiveSwap {
...
+        let swap_info = self
+            .persister
+            .get_swap_info_by_address(req.swap_address.clone())?
+            .ok_or_else(|| anyhow!(format!("swap address {} was not found", req.swap_address)))?;
+
+        let transactions = self
+            .chain_service
+            .address_transactions(req.swap_address.clone())
+            .await?;
+        let utxos = get_utxos(req.swap_address, transactions)?;
```
Comment (roeierez) [inline libs/sdk-core/src/swap.rs:450]: Perhaps we can extract out this code to a function that can be reused in `refund_swap` function?

**Review by roeierez — CHANGES_REQUESTED** [review]
Thanks a lot, looks good!
Can you please make sure the RN generator code is done with the latest RN generator? We yesterday updated it.

File: libs/sdk-core/src/swap.rs (hunk 1 above)
Comment (danielgranhao) [reply libs/sdk-core/src/swap.rs]: Makes sense. Addressed in [a2308ba](https://github.com/breez/breez-sdk/pull/548/commits/a2308ba88b4b60bc483a1187c071ce5224624c6b)

**Comment by danielgranhao** [comment]
@roeierez 
> Can you please make sure the RN generator code is done with the latest RN generator? We yesterday updated it.

I've just rebased and run the generator again. I didn't see any changes.

**Review by dangeross — APPROVED** [review]
RN codegen looks good

**Comment by roeierez** [comment]
@danielgranhao this is ready to be merged. We will merge it right after the conflict is resolved. Thanks.

**Comment by danielgranhao** [comment]
@roeierez Conflicts should be resolved now :)

**Pull request merged**